packages/vite/src/node/server/index.ts

```diff
@@ -829,7 +832,11 @@ async function restartServer(server: ViteDevServer) {
   const shortcutsOptions = server._shortcutsOptions
   const oldUrls = server.resolvedUrls
 
-  let inlineConfig = server.config.inlineConfig
+  let inlineConfig = mergeConfig(server.config.inlineConfig, {
+    server: {
+      port: _currentServerPort,
```

Comment:
The issue is that it could override a change in the config port which would in that case be expected to switch port. Does it work to add a currentServerPort check just before `await server.listen(port, true)`?

Reply:
Yes, this is indeed a problem. 👍 
It seems to interfere with the config configuration. I will consider how to deal with it.